--- provider/postgresql.py	2021-12-01 00:00:00.000000000 +0000
+++ provider/postgresql.py	2021-12-01 00:00:00.000000000 +0000
@@ -302,7 +302,7 @@
             orderby = self._make_orderby(sortby) if sortby else SQL('')

             sql_query = SQL("DECLARE \"geo_cursor\" CURSOR FOR \
-             SELECT DISTINCT {} {} FROM {} {} {}").\
+             SELECT {} {} FROM {} {} {}").\
                 format(props,
                        geom,
                        Identifier(self.table),
